stdlib/public/stubs/UnicodeShims.cpp

```diff
@@ -1,61 +0,0 @@
-//===--- UnicodeNormalization.cpp - Unicode Normalization Helpers ---------===//
-//
-// This source file is part of the Swift.org open source project
-//
-// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
-// Licensed under Apache License v2.0 with Runtime Library Exception
-//
-// See https://swift.org/LICENSE.txt for license information
-// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
-//
-//===----------------------------------------------------------------------===//
-//
-// Functions that use ICU to do unicode normalization and collation.
-//
-//===----------------------------------------------------------------------===//
-
-#include "../SwiftShims/UnicodeShims.h"
-
-#include <stdint.h>
-
-#if defined(__APPLE__)
-
-// Declare a few external functions to avoid a dependency on ICU headers.
-extern "C" {
-
-// Types
-typedef struct UBreakIterator UBreakIterator;
-typedef enum UErrorCode {} UErrorCode;
-typedef enum UCharNameChoice {} UCharNameChoice;
-typedef uint16_t UChar;
-typedef int32_t UChar32;
-typedef int8_t UBool;
-typedef __swift_stdlib_UProperty UProperty;
-
-#define U_MAX_VERSION_LENGTH 4
-typedef uint8_t UVersionInfo[U_MAX_VERSION_LENGTH];
-}
-
-#else
-
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdocumentation"
-
-#include <unicode/ustring.h>
-#include <unicode/ucol.h>
-#include <unicode/ucoleitr.h>
-#include <unicode/uiter.h>
-#include <unicode/uchar.h>
-#include <unicode/ustring.h>
-#include <unicode/uvernum.h>
-#include <unicode/uversion.h>
-
-#pragma clang diagnostic pop
-
-#endif
-
-// Force an autolink with ICU
-#if defined(__MACH__)
-asm(".linker_option \"-licucore\"\n");
-#endif // defined(__MACH__)
```

Comment:
If the stdlib isn't doing this anymore, would importing Foundation need to?

Reply:
Foundation would, and already does so by means of the linker script.  The need for this is strictly for static linking.  When statically linking, the dependencies must be explicitly listed and this is a horrible workaround to avoid the user from having to specify that and the driver not learning the dependency.  This can be safely removed IMO.

Reply:
https://github.com/apple/swift-corelibs-foundation/blob/main/Sources/Foundation/CMakeLists.txt#L171-L182 for the hackery that I was referring to - the dependency is injected via the swift driver at compile time.